tools/codegen/gen_lazy_tensor.py

```diff
@@ -144,7 +172,15 @@ def gen_key(func: FunctionSchema) -> Tuple[str, str]:
     if impl_path is not None:
         error_on_missing_kernels(native_functions, backend_indices, backend_key,
                                  autograd_key, impl_path, full_codegen)
-
+    if shape_inference_hdr is not None:
+        expected_shape_infr_decls = list(
+            concat_map_codegen(
+                dest.GenLazyShapeInferenceDefinition(backend_indices[backend_key], tensor_class),
```

Comment:
Do you mind writing out somewhere what the flow looks like for non-meta tensor shape inference? It'd help to have it listed somewhere:
(1) what all of the components are (the headers, shape inference kernels, their call-sites)
(2) which parts are codegen'd